lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/metrics/Histogram.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.base.metrics;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.stats.TDigest;
+import io.trino.spi.metrics.Metric;
+
+public class Histogram
+        implements Metric<Histogram>
+{
+    private final TDigest digest;
+
+    @JsonCreator
+    public Histogram(TDigest digest)
+    {
+        this.digest = digest;
+    }
+
+    @JsonProperty("digest")
+    public TDigest getDigest()
```

Comment:
TDigest is not json-serializable.

Reply:
Oops, you're right - added a custom converter to serialize/deserialize TDigest (using base64).